### PR TITLE
Safer dependencies

### PR DESCRIPTION
**Pull request by bmarty**
Ref:
- https://docs.gradle.org/5.1.1/userguide/declaring_repositories.html#sec::matching_repositories_to_dependencies
- https://jebware.com/blog/?p=573

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -37,45 +37,198 @@ allprojects {
...
+                includeGroupByRegex 'com\\.android\\.tools\\..*'
+                includeGroupByRegex "com\\.google\\.android\\..*"
+                includeGroup 'com.google.firebase'
+                includeGroup 'com.android'
+                includeGroup 'com.android.tools'
+            }
+        }
+        mavenCentral {
+            content {
+                includeGroup 'com.adevinta.android'
```
Comment (bmarty) [inline build.gradle:84]: if only we had a way to externalize those list into another file...
`includeGroup` does not support taking a list as param

Comment (ariskotsomitopoulos) [reply build.gradle:84]: let me check if that is possible

**Comment by bmarty** [comment]
@ariskotsomitopoulos thanks for your proposal solution. I have updated it a bit. Please let me know what you think.

**Review by ariskotsomitopoulos — APPROVED** [review]

**Comment by ariskotsomitopoulos** [comment]
> @ariskotsomitopoulos thanks for your proposal solution. I have updated it a bit. Please let me know what you think.

Yes +1, nice changes! I agree

File: vector/build.gradle [hunk 2]
```diff
@@ -317,11 +317,6 @@ android {
     }
 }
 
-configurations {
-    // videocache includes a sl4j logger which causes mockk to attempt to call the static android Log
-    testImplementation.exclude group: 'org.slf4j', module: 'slf4j-android'
-}
```
Comment (bmarty) [inline vector/build.gradle:323]: As I removed `videocache`, I thought I could remove this as well. But maybe double check with @ouchadam first. And I just see that it's still needed: https://github.com/vector-im/element-android/pull/4670/files#diff-f6dc61a00090b6fc204667d7934e656dbb78c8bb0dc14f95468dd57b4300a8caR177

File: build.gradle [hunk 3]
```diff
@@ -37,45 +36,50 @@ allprojects {
...
-                includeGroupByRegex 'com\\.github\\.jetradarmobile'
-                includeGroupByRegex 'nl\\.dionsegijn'
-
-                // Voice RecordView
-                includeGroupByRegex 'com\\.github\\.Armen101'
+                groups.jitpack.regex.each { includeGroupByRegex it }
+                groups.jitpack.group.each { includeGroup it }
             }
         }
-        maven { url 'https://oss.sonatype.org/content/repositories/snapshots/' }
```
Comment (bmarty) [inline build.gradle:69]: I am not sure why it has been added at first place, but it seems not necessary anymore 🤞

File: vector/build.gradle (hunk 2 above)
Comment (ouchadam) [reply vector/build.gradle:323]: if `videocache` is removed then removing this is also fine :+1:  (the unit tests will fail if not)

**Review by ouchadam — APPROVED** [review]
great change! :100: (might even have a little bit of a build speed improvement :crossed_fingers: )

**Pull request merged**